pwiz_tools/Skyline/FileUI/PeptideSearch/ImportPeptideSearchDlg.cs

```diff
@@ -1043,6 +1037,10 @@ public void CloseWizard(DialogResult result)
         {
             // Close file handles to the peptide search library
             ImportPeptideSearch.ClosePeptideSearchLibraryStreams(Document);
+
+            // Dispose DDA SearchEngine if it is initialized
+            ImportPeptideSearch.SearchEngine?.Dispose();

```

Comment:
This does not seem like the right place to put this line.
If the user closes the wizard by clicking the X button on the dialog, I think this code will not run.

Usually we put cleanup code like this in OnFormClosed.

Reply:
Does the same logic also apply to ClosePeptideSearchLibraryStreams? But I agree it should definitely run if the user closes the wizard, so does Kaipo's suggestion work?

Reply:
I guess I was assuming that the Cancel button and the X button would lead to the same result (why wouldn't they, at least from the user's perspective?)

Reply:
Yes, ClosePeptideSearchLibraryStreams should go in OnFormClosed as well.

CloseWizard really should just be the one line:
DialogResult = result;
(or we could get rid of the function entirely and have the caller do that)

This code was written in 2012, and I think we did not know what we were doing back then.